### PR TITLE
Consume hcaptcha and pwn deps

### PR DESCRIPTION
**Pull request by jolheiser**
This PR just consumes the [hcaptcha](https://gitea.com/jolheiser/hcaptcha) and [haveibeenpwned](https://gitea.com/jolheiser/pwn) modules directly into Gitea.

Nothing else uses them, so I figured they would be better off in the main repo rather than sitting stale in another repo.

Bringing them into the main repo will help keep eyes on them and also notify us if any of the testing starts failing (as I noticed the hcaptcha testing has changed since I made the original module for secret keys).

-----

Note that the modules were made to be libraries, so copying them here more than likely has some more code than is needed directly by Gitea. 

I'm not opposed to trimming it down, but as a first pass I just wanted to get them in here.

---

Also let this serve as a notice that I'm fine with transferring my license (which was already MIT) from my own name to "The Gitea Authors".

### CODE REVIEW

File: modules/password/pwn/pwn_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		t.Logf("%s should fail as a password\n", failPad)
+		t.Fail()
+	}
+
+	// Checking for a "good" password isn't going to be perfect, but we can give it a good try
+	// with hopefully minimal error. Try five times?
+	var good bool
+	var pw string
+	for idx := 0; idx <= 5; idx++ {
+		pw = testPassword()
```
Comment (jolheiser) [inline modules/password/pwn/pwn_test.go:66]: This wasn't an ideal test then, and it isn't an ideal test now. Open to suggestions for this.

**Review by zeripath — APPROVED** [review]

**Comment by lunny** [comment]
I think we should clarify which modules we should merge into Gitea and which we should depends on third-party on documentation. My idea is depending on third-party moudules ASAP if it's mature and well mantained.

**Comment by jolheiser** [comment]
I briefly looked, but admittedly not very hard, as these are fairly simple APIs.

That being said, these are largely equivalent to their module counterpart. I just wanted to increase bus factor, I'm not against swapping them in the future if we find a better library elsewhere.

**Review by lunny — APPROVED** [review]

**Comment by jolheiser** [comment]
:trumpet: :robot:

**Pull request merged**